hackathon_site/event/test_api.py

```diff
@@ -366,3 +366,136 @@ def test_name_search_filter(self):
         results = data["results"]
         returned_ids = [res["team_code"] for res in results]
         self.assertCountEqual(returned_ids, [self.team2.team_code])
+
+
+class ProfileDetailViewTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.user_with_profile = User.objects.create_user(
+            username="foo2@bar.com",
+            password=self.password,
+            first_name="Test2",
+            last_name="Bar2",
+            email="foo2@bar.com",
+        )
+        self.profile = self._make_event_profile(user=self.user_with_profile)
+        self.request_body = {
+            "id_provided": True,
+            "attended": True,
+            "acknowledge_rules": True,
+            "e_signature": "user signature",
+        }
+        self.change_permissions = Permission.objects.filter(
+            content_type__app_label="event", codename="change_profile"
+        )
+        self.expected_response = {
+            "id": self.profile.id,
+            "id_provided": True,
+            "attended": True,
+            "acknowledge_rules": True,
+            "e_signature": "user signature",
+            "team": self.profile.team_id,
+        }
+        self.view = reverse("api:event:profile-detail", kwargs={"pk": self.profile.id})
+
+    def test_user_not_logged_in(self):
+        response = self.client.patch(self.view, self.request_body)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_has_no_change_permissions(self):
+        self._login()
+        response = self.client.patch(self.view, self.request_body)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_user_has_change_permissions(self):
+        self._login(self.change_permissions)
+        response = self.client.patch(self.view, self.request_body)
+        data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.expected_response, data)
+
+    def test_modifying_acknowledge_rules_and_e_signature_twice(self):
+        self._login(self.change_permissions)
+
+        # First, normally update profile
+        response = self.client.patch(self.view, self.request_body)
+        data = response.json()
+        expected_response = self.expected_response
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(expected_response, data)
+
+        # Then update profile with a different request, changing acknowledge & e signature
+        new_request_body = {
```

Comment:
One test you'll want to have: What if they only provide a single field? I think the current code will crash.